### PR TITLE
Reverting contact no encryption and laying the new encryption system

### PR DESCRIPTION
**Pull request by ThinuwanW**
…t-on-read

### Summary

-

### How to test

1.

### Project Checklist

- [ ] Changes build without any errors
- [ ] Have written adequate test cases
- [ ] Done developer testing in
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
- [ ] Code is formatted with `npm run format`
- [ ] Code is linted with `npm run check-lint`
- [ ] No unnecessary comments left in code
- [ ] Made corresponding changes to the documentation

#### Other

- [ ] New atomic components added
- [ ] New molecules added
- [ ] New pages(routes) added
- [ ] New dependencies installed

### PR Checklist

- [ ] Pull request is raised from the correct source branch
- [ ] Pull request is raised to the correct destination branch
- [ ] Pull request is raised with correct title
- [ ] Pull request is self reviewed
- [ ] Pull request is self assigned
- [ ] Suitable pull request status labels are added (`ready-for-code-review`)

### Additional Information
-

### CODE REVIEW

**Review by ThinuwanW — COMMENTED** [review]
## 🤖 Claude Code Review

This PR adds a transitional JPA AttributeConverter (LegacyEncryptedContactNoConverter) that stops encrypting the organization contact_no on write while still decrypting legacy "ENC:" values on read, paired with the EpOrganization entity switching to it. The converter logic is correct and faithful: null-safe, the decrypt-on-read failure path is already wrapped in ModuleException inside EncryptionDecryptionServiceImpl, and the static setter-injection pattern matches the existing EncryptionDecryptionConverter. The main gaps are the absence of unit tests for the new behavior and some duplication/placement considerations given its transitional nature.

Found **5** new issue(s): 🟣 2 nit(s)

File: backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package com.skapp.community.common.util.converter;
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java:1]: 🤖 **Claude** · 💬 **Note**

📁 **File-level observation**

Missing unit tests for the new converter. The sibling EncryptionDecryptionConverter has a dedicated test class (EncryptionDecryptionConverterTest), but this converter introduces distinct behavior that is currently unverified: convertToDatabaseColumn must now return the value unchanged with NO "ENC:" prefix (encryption stopped), while convertToEntityAttribute must still (a) decrypt legacy "ENC:" values, (b) return plaintext as-is, and (c) handle null. These are exactly the behaviors the safe rollout depends on. Add a LegacyEncryptedContactNoConverterTest mirroring the existing test that asserts plaintext-on-write and decrypt-on-read for legacy values.

File: backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	 * migration has converted every stored value back to plaintext.
+	 */
+	@Override
+	public String convertToEntityAttribute(String dbData) {
+		if (dbData == null) {
+			return null;
+		}
+		if (dbData.startsWith(ENCRYPTED_PREFIX)) {
+			return encryptionDecryptionService.decrypt(dbData.substring(ENCRYPTED_PREFIX.length()));
+		}
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java:49]: 🤖 **Claude** · 💬 **Note**

convertToEntityAttribute, the ENCRYPTED_PREFIX constant, and the static EncryptionDecryptionService setter-injection are near-identical copies of EncryptionDecryptionConverter. Given the class is explicitly transitional/disposable this duplication is defensible, but consider extracting the shared decrypt-on-read helper (or at least the "ENC:" prefix constant) so the two converters cannot drift apart while both exist. Minor — acceptable to leave if the migration/removal lands soon.

File: backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ */
+@Converter
+@Component
+public class LegacyEncryptedContactNoConverter implements AttributeConverter<String, String> {
+
+	// Prefix that marks values written by the previous (encrypting) converter.
+	private static final String ENCRYPTED_PREFIX = "ENC:";
+
+	private static EncryptionDecryptionService encryptionDecryptionService;
+
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java:23]: 🤖 **Claude** · 💬 **Note**

Package/module placement: this converter exists solely to support an enterprise-only field (EpOrganization.contactNo) yet lives in the community module. It works because enterprise may depend on community and it only references the community EncryptionDecryptionService, and it is consistent with the existing converter's location, so this is a nit. If the intent is to keep enterprise-specific transitional code out of community, it could instead live under com.skapp.enterprise; otherwise leaving it here is fine.

File: backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+ * revert migration in skapp-migrations.
+ * </p>
+ *
+ * @see EncryptionDecryptionConverter the generic encrypt-at-rest converter, retained for
+ * fields that genuinely require field-level encryption
+ */
+@Converter
+@Component
+public class LegacyEncryptedContactNoConverter implements AttributeConverter<String, String> {
+
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java:18]: 🤖 **Claude** · 🟣 **Nit**

Malformed Javadoc: `@see EncryptionDecryptionConverter the generic encrypt-at-rest converter...` — the class-reference form of `@see` does not take trailing label prose, so the trailing text is treated as part of the reference and will render incorrectly / may produce a javadoc warning. Use a separate sentence with `{@link EncryptionDecryptionConverter}` instead, e.g. a normal paragraph referencing `{@link EncryptionDecryptionConverter}` for fields that genuinely require field-level encryption. Severity: nit.

File: backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java (hunk 1 above)
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java:1]: 🤖 **Claude** · 🟣 **Nit**

📁 **File-level observation**

Paired-change dependency not verifiable in-repo: correctness of this converter (and the eventual safe removal of the decrypt-on-read branch) hinges on the referenced "GDPR contact-number revert migration in skapp-migrations", which lives in a separate repository. There is no Liquibase changeset or in-repo guard ensuring that migration ships before/with this change, nor preventing premature removal of this converter while legacy ENC: rows still exist. Confirm the external migration is coordinated with this deploy, and consider adding a brief note/ticket reference tying the converter's removal to migration completion so a future cleanup doesn't render legacy rows unreadable. Severity: nit.

Comment (ThinuwanW) [reply backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java:1]: no need of ticket and anything since both this change and the migration will rollout together and the whole migration implementation is there in the migration repo pr

File: backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java (hunk 3 above)
Comment (ThinuwanW) [reply backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java:23]: true but keeping this way under following reasons
1. existing code is also in the community so this is something more same as it
2. this and the existing code both will remove after completing the migration

File: backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java (hunk 4 above)
Comment (ThinuwanW) [reply backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java:18]: this is valid and AI generated comment was malformed.
fixed.

File: backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java (hunk 2 above)
Comment (ThinuwanW) [reply backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java:49]: valid
but this file added while keeping the existing as it is and this is a temp code which will remove soon

[the remark this answers is not in the document]
Comment (ThinuwanW) [reply backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java]: valid point but this is used in contact number where this is not possible. in the new implementation I'll add a guard and not in this since this is a temp code

File: backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+	 */
+	@Override
+	public String convertToEntityAttribute(String dbData) {
+		if (dbData == null) {
+			return null;
+		}
+		if (dbData.startsWith(ENCRYPTED_PREFIX)) {
+			return encryptionDecryptionService.decrypt(dbData.substring(ENCRYPTED_PREFIX.length()));
+		}
+		return dbData;
```
[the remark this answers is not in the document]
Comment (ThinuwanW) [reply backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java:50]: added

File: backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java (hunk 1 above)
Comment (ThinuwanW) [reply backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java:1]: added

File: backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+ * (and ultimately this whole converter) can be removed — see the GDPR contact-number
+ * revert migration in skapp-migrations.
+ * </p>
+ *
+ * @see EncryptionDecryptionConverter the generic encrypt-at-rest converter, retained for
+ * fields that genuinely require field-level encryption
+ */
+@Converter
+@Component
+public class LegacyEncryptedContactNoConverter implements AttributeConverter<String, String> {
```
[the remark this answers is not in the document]
Comment (ThinuwanW) [reply backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java:17]: this is used in submodule so we can take this as an invalid comment

File: backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
+package com.skapp.community.common.util.converter;
+
+import com.skapp.community.common.service.EncryptionDecryptionService;
+import jakarta.persistence.AttributeConverter;
+import jakarta.persistence.Converter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Transitional JPA AttributeConverter for the organization {@code contact_no} field.
```
[the remark this answers is not in the document]
Comment (ThinuwanW) [reply backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java]: not valid 
submodule probem

**Review by ThinuwanW — COMMENTED** [review]
## 🤖 Claude Code Review

This PR adds an InternalServerException type (HTTP 500) with cause-chain unwrapping in GlobalExceptionHandler to surface coded server-side faults that Hibernate rewraps during flush/load, plus a transitional LegacyEncryptedContactNoConverter that stops encrypting contact_no on write while still decrypting legacy ENC: values on read. The changes are clean, consistent with existing patterns, and well-tested. Findings are minor: an asymmetry where the legacy converter's decrypt failures aren't covered by the new unwrap mechanism, code duplication between the two coded-exception types, and an unguarded cause-chain walk.

Found **6** new issue(s): 
_1 finding(s) already posted in a previous review._

File: backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+	 * "ENC:" prefix, so the application keeps reading correct values until the data
+	 * migration has converted every stored value back to plaintext.
+	 */
+	@Override
+	public String convertToEntityAttribute(String dbData) {
+		if (dbData == null) {
+			return null;
+		}
+		if (dbData.startsWith(ENCRYPTED_PREFIX)) {
+			return encryptionDecryptionService.decrypt(dbData.substring(ENCRYPTED_PREFIX.length()));
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java:48]: 🤖 **Claude** · 💬 **Note**

This converter calls the community EncryptionDecryptionService.decrypt(), which throws ModuleException (mapped to HTTP 400) on failure — not InternalServerException. Since this runs inside Hibernate result-set mapping, a decrypt failure on a legacy ENC: row gets rewrapped as a DataAccessException, and the new findInternalServerException() unwrap only looks for InternalServerException. Net effect: the coded COMMON_ERROR_DECRYPTION_FAILED message is never surfaced — the client gets a generic COMMON_ERROR_DATABASE_ERROR 500, and semantically a server-side decrypt fault is being modeled as a client error (400). Consider wrapping decrypt failures here in InternalServerException so this path benefits from the very unwrap mechanism this PR adds, keeping it symmetric with the enterprise FieldEncryptionService path. Low impact since this is transitional code, but the infrastructure added doesn't actually cover this converter.

File: backend/src/main/java/com/skapp/community/common/exception/InternalServerException.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
+package com.skapp.community.common.exception;
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/common/exception/InternalServerException.java:1]: 🤖 **Claude** · 💬 **Note**

📁 **File-level observation**

InternalServerException is a near-verbatim copy of ModuleException: same static AtomicReference<MessageUtil>, identical getMessageUtil() with the 'MessageUtil not initialized' guard, and a duplicate @Component MessageUtilInjector ApplicationContextAware inner class (now two beans doing the same MessageUtil wiring into separate static refs). Consider extracting the shared machinery into a common abstract base (e.g. AbstractCodedException holding messageKey + the MessageUtil lookup) so the two coded-exception types differ only in their HTTP-status semantics. Reduces the maintenance burden of keeping two injectors in sync.

File: backend/src/main/java/com/skapp/community/common/exception/GlobalExceptionHandler.java [hunk 10]
```diff
@@ -177,6 +203,20 @@ public ResponseEntity<ResponseEntityDto> handleExceptions(Exception e) {
 				status);
 	}
 
+	/**
+	 * Walks the cause chain looking for an {@link InternalServerException}. Hibernate and
+	 * Spring wrap exceptions thrown during flush/load, so the original is rarely the
+	 * top-level type seen by these handlers.
+	 */
+	private InternalServerException findInternalServerException(Throwable throwable) {
+		for (Throwable current = throwable; current != null; current = current.getCause()) {
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/common/exception/GlobalExceptionHandler.java]: 🤖 **Claude** · 💬 **Note**

findInternalServerException() walks getCause() with no cycle/visited guard. A self-referential or cyclic cause chain would loop forever (vs. Spring's NestedExceptionUtils-style walkers that are cycle-safe). Cyclic chains are rare in practice, but consider a bounded loop or a visited set to be defensive, since this runs on every DataAccessException and unhandled Exception.

File: backend/src/main/java/com/skapp/community/common/exception/GlobalExceptionHandler.java [hunk 11]
```diff
@@ -157,6 +166,15 @@ public ResponseEntity<ResponseEntityDto> handleSQLException(SQLException e) {
 
 	@ExceptionHandler(DataAccessException.class)
 	public ResponseEntity<ResponseEntityDto> handleDataAccessException(DataAccessException e) {
+		// A field-encryption (or other server-side) fault thrown inside a JPA flush/load
+		// is
+		// rewrapped by Hibernate as a DataAccessException — unwrap it so the original,
+		// coded error surfaces instead of a generic database error.
+		InternalServerException internalServerException = findInternalServerException(e);
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/common/exception/GlobalExceptionHandler.java]: 🤖 **Claude** · 💬 **Note**

The new unwrap helper recovers only InternalServerException from a Hibernate/Spring-wrapped cause chain — ModuleException is not unwrapped. This is asymmetric and directly bites the new LegacyEncryptedContactNoConverter: its convertToEntityAttribute() calls the community EncryptionDecryptionService.decrypt(), which throws ModuleException on a corrupt legacy 'ENC:' value during a JPA load. When Hibernate rewraps that into a DataAccessException, findInternalServerException() returns null and the response degrades to a generic 500 COMMON_ERROR_DATABASE_ERROR, losing the coded decryption message. Either also walk the chain for ModuleException (returning its mapped 400/coded response) or align the converter to throw InternalServerException so the intended coded 500 surfaces. As written the unwrap only helps the enterprise FieldEncryption path, not the converter this PR adds.

File: backend/src/test/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverterTest.java [hunk 12]
```diff
@@ -0,0 +1,104 @@
+package com.skapp.community.common.util.converter;
```
Comment (ThinuwanW) [inline backend/src/test/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverterTest.java:1]: 🤖 **Claude** · 💬 **Note**

📁 **File-level observation**

The converter's entire reason to exist is safely handling legacy encrypted values on read, yet the test only exercises happy paths (plaintext passthrough and successful legacy decrypt). There is no test for a value that carries the 'ENC:' prefix but whose payload is corrupt/non-Base64 — i.e. the failure path that motivated this PR's exception-handling changes. Add a test asserting the expected exception/behavior for a malformed 'ENC:'-prefixed value, so the decrypt-failure contract (and any future change to which exception type is thrown) is locked down.

File: backend/pom.xml [hunk 13]
```diff
@@ -283,6 +283,17 @@
             <artifactId>cloudfront</artifactId>
             <version>2.42.0</version>
         </dependency>
+        <dependency>
+            <groupId>software.amazon.awssdk</groupId>
+            <artifactId>kms</artifactId>
+            <version>2.42.0</version>
```
Comment (ThinuwanW) [inline backend/pom.xml:289]: 🤖 **Claude** · 💬 **Note**

The new KMS dependency hardcodes version 2.42.0, the third software.amazon.awssdk artifact to repeat that literal (alongside the existing s3/cloudfront entries) with no shared version. This invites version drift across AWS SDK modules. Consider importing software.amazon.awssdk:bom in <dependencyManagement> (as is already done for azure-sdk-bom) and dropping the per-artifact <version>, so all AWS SDK artifacts stay aligned. Nit — consistent with current style, but the BOM avoids future skew.

Comment (ThinuwanW) [reply backend/pom.xml:289]: already noted and is a separate improvement

File: backend/src/test/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverterTest.java (hunk 12 above)
Comment (ThinuwanW) [reply backend/src/test/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverterTest.java:1]: this operation is actively monitored and temp code so no need for edge cases

File: backend/src/main/java/com/skapp/community/common/exception/GlobalExceptionHandler.java (hunk 11 above)
Comment (ThinuwanW) [reply backend/src/main/java/com/skapp/community/common/exception/GlobalExceptionHandler.java]: LegacyEncryptedContactNoConverter is temp code and have only basic exception handling

File: backend/src/main/java/com/skapp/community/common/exception/GlobalExceptionHandler.java (hunk 10 above)
Comment (ThinuwanW) [reply backend/src/main/java/com/skapp/community/common/exception/GlobalExceptionHandler.java]: This new implementation is unnecessary for the existing code so reverted and kept the original way

File: backend/src/main/java/com/skapp/community/common/exception/InternalServerException.java (hunk 9 above)
Comment (ThinuwanW) [reply backend/src/main/java/com/skapp/community/common/exception/InternalServerException.java:1]: True but lets keep the pattern as it is and take as a separate improvement

File: backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java (hunk 8 above)
Comment (ThinuwanW) [reply backend/src/main/java/com/skapp/community/common/util/converter/LegacyEncryptedContactNoConverter.java:48]: skipping since the temp code

**Review by akilarootcode — APPROVED** [review]
